pallets/identity/src/lib.rs

```diff
@@ -259,25 +269,38 @@ decl_module! {
             };
             use polymesh_common_utilities::traits::identity::runtime_upgrade::LinkedKeyInfo;
 
-            migrate_map::<LinkedKeyInfo, _>(
-                b"identity",
-                b"KeyToIdentityIds",
-                |_| Empty
-            );
-            // Migrate secondary key permissions to the new type
-            migrate_map::<IdentityWithRolesOld<T::AccountId>, _>(
-                b"identity",
-                b"DidRecords",
-                |_| Empty
-            );
-            // Remove roles from Identities
-            StorageIterator::<IdentityWithRoles<T::AccountId>>::new(b"identity", b"DidRecords")
-                .drain()
-                .map(|(key, old)|  (key, DidRecord {
-                    primary_key: old.primary_key,
-                    secondary_keys: old.secondary_keys,
-                }))
+            let storage_ver = <StorageVersion>::get();
+
+            // Migrate from V0 to V1
+            if storage_ver < Version::V1 {
```

Comment:
we can make it `==`

Reply:
Using '<' operator allows us to load `<StorageVersion>` just once.

Reply:
At any rate, it seems most consistent to use the same `<` operator uniformly.

Reply:
The new macro `storage_migrate_on!` fixes this.